pkg/certificate/providers/vault/certificate_manager.go

```diff
@@ -30,6 +30,21 @@ const (
 
 // New constructs a new certificate client using Vault's cert-manager
 func New(vaultAddr, token, role string) (*CertManager, error) {
```

Comment:
not a big deal, but you could keep these as one function

Reply:
I split this up in order to make the client creation (without adding the root certificate) testable. Do you have any suggestions for how to keep this a single function without requiring too much refactoring of the tests?

Reply:
oh i see. Ya you can leave as is.

Alternatively you could do something like

```
// package level
type getRootCAFunc func(c *CertManager)
var getRootCA = func() {...} // fill in with current getRootCa (not a function not a method)
```

then in tests, before running:

```
getRootCA = func(){...} // overridden for testing
```

Up to you!